### PR TITLE
add support of IAM role for get

### PR DESCRIPTION
**Pull request by DR3EVR8u8c**
Thanks for your work. I enabled the IAM roles support and wondering someone else may need that too.

### CODE REVIEW

**Comment by DR3EVR8u8c** [comment]
Sorry, I'm confused. I saw the check is passed and I am happy to merge it into the master. But, do I need to do anything, or, it has been done ealready?
I am new to GitHub. not sure how to merger.

File: z3/snap.py [hunk 1]
```diff
@@ -573,17 +573,15 @@ def main():
...
-            extra_config['host'] = cfg['HOST']
-    except KeyError as err:
-        sys.stderr.write("Configuration error! {} is not set.\n".format(err))
-        sys.exit(1)
-
-    bucket = boto.connect_s3(s3_key_id, s3_secret, **extra_config).get_bucket(bucket)
+        bucket = boto.connect_s3(s3_key_id, s3_secret, **extra_config).get_bucket(bucket)
+    else:
+        bucket = cfg['BUCKET']
+        bucket = boto.connect_s3( **extra_config).get_bucket(bucket)
```
Comment (vtemian) [inline z3/snap.py]: There is an extra ` `.

File: z3/pput.py [hunk 2]
```diff
@@ -332,8 +332,12 @@ def main():
     if 'HOST' in CFG:
         extra_config['host'] = CFG['HOST']
 
-    bucket = boto.connect_s3(
-        CFG['S3_KEY_ID'], CFG['S3_SECRET'], **extra_config).get_bucket(CFG['BUCKET'])
+    if 'S3_KEY_ID' in CFG:
+        bucket = boto.connect_s3(
+            CFG['S3_KEY_ID'], CFG['S3_SECRET'], **extra_config).get_bucket(CFG['BUCKET'])
+    else:
+        bucket = boto.connect_s3(
```
[the remark this answers is not in the document]
Comment (DR3EVR8u8c) [reply z3/pput.py]: boto3 s3 check bucket automatically. if bucket is not existed , it will raise 404 error

File: z3/snap.py (hunk 1 above)
Comment (DR3EVR8u8c) [reply z3/snap.py]: I don't see much use of host parameter for accessing s3. I assume AWS will choose the best path. Also, is region a better option when comparing with static endpoint?

Comment (DR3EVR8u8c) [reply z3/snap.py]: I agree extra config is useful. I will test it further and commit it again

Comment (DR3EVR8u8c) [reply z3/snap.py]: add extra_config support, tested fine

File: z3/pput.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vtemian) [reply z3/pput.py]: My concern was regarding the `CFG['BUCKET']`. You can use `CFG.get('BUCKET')` or 
```python
if not 'BUCKET' in CFG:
    raise ValueError('You need to define a bucket on which to upload.')
```

File: z3/snap.py (hunk 1 above)
Comment (vtemian) [reply z3/snap.py]: 👍

File: z3/pput.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vtemian) [reply z3/pput.py]: Since there wasn't any check before, let's merge it.

**Pull request merged**